api/admin/controllers/site.js

```diff
@@ -69,4 +73,25 @@ module.exports = wrapHandlers({
 
     return res.json(serializeNew(site, true));
   },
+
+  destroy: async (req, res) => {
+    let site;
+    const { id } = req.params;
+
+    try {
+      site = await fetchSiteById(id);
+      await S3SiteRemover.removeSite(site);
+      await S3SiteRemover.removeInfrastructure(site);
+
+      if (Features.enabled(Features.Flags.FEATURE_PROXY_EDGE_DYNAMO)) {
+        await ProxyDataSync.removeSite(site);
+      }
+
+      await site.destroy();
+      return res.json(serializeNew(site, true));
+    } catch (error) {
+      logger.error([`site@id=${site.id}`, error.message, error.stack].join('\n\n'));
```

Comment:
I'm not a huge fan of putting these messages in the logs since no one ever sees them. We need to find a better way of surfacing when these things happen...

Reply:
Put a `To Do` until we role out events.